File: mobile/lib/models/service_material.dart
class ServiceMaterial {
  final int id;
  final String name;
  final String? brand;
  final int quantity;
  final String unit;
  final double? estimatedPrice;
  final String? notes;

  ServiceMaterial({
    required this.id,
    required this.name,
    this.brand,
    required this.quantity,
    required this.unit,
    this.estimatedPrice,
    this.notes,
  });

  factory ServiceMaterial.fromJson(Map<String, dynamic> json) {
    return ServiceMaterial(
      id: json['id'],
      name: json['name'],
      brand: json['brand'],
      quantity: json['quantity'],
      unit: json['unit'],
      estimatedPrice: json['estimatedPrice']?.toDouble(),
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'brand': brand,
      'quantity': quantity,
      'unit': unit,
      'estimatedPrice': estimatedPrice,
      'notes': notes,
    };
  }
}

File: api/AppEAO.API/Domain/Entities/ServiceMaterial.cs
namespace AppEAO.API.Domain.Entities;

public class ServiceMaterial
{
    public int Id { get; private set; }
    public int ServiceId { get; private set; }
    public string Name { get; private set; } = string.Empty;
    public string? Brand { get; private set; }
    public int Quantity { get; private set; }
    public string Unit { get; private set; } = string.Empty; // unidade, kg, m², etc
    public decimal? EstimatedPrice { get; private set; }
    public string? Notes { get; private set; }
    public DateTime CreatedAt { get; private set; }
    public DateTime UpdatedAt { get; private set; }
    public bool IsActive { get; private set; }

    // Navigation properties
    public Service Service { get; private set; } = null!;

    // Constructor for EF Core
    private ServiceMaterial() { }

    public ServiceMaterial(
        int serviceId,
        string name,
        int quantity,
        string unit,
        string? brand = null,
        decimal? estimatedPrice = null,
        string? notes = null)
    {
        ServiceId = serviceId;
        Name = name;
        Brand = brand;
        Quantity = quantity;
        Unit = unit;
        EstimatedPrice = estimatedPrice;
        Notes = notes;
        CreatedAt = DateTime.UtcNow;
        UpdatedAt = DateTime.UtcNow;
        IsActive = true;
    }

    public void Update(
        string name,
        int quantity,
        string unit,
        string? brand = null,
        decimal? estimatedPrice = null,
        string? notes = null)
    {
        Name = name;
        Brand = brand;
        Quantity = quantity;
        Unit = unit;
        EstimatedPrice = estimatedPrice;
        Notes = notes;
        UpdatedAt = DateTime.UtcNow;
    }

    public void Deactivate()
    {
        IsActive = false;
        UpdatedAt = DateTime.UtcNow;
    }
}
